engine/src/connector.rs

```diff
@@ -10,61 +10,68 @@ use crate::parameters::{
 };
 use crate::proof::Proof;
 use aurora_engine_sdk::env::Env;
 use aurora_engine_sdk::io::{StorageIntermediate, IO};
 use aurora_engine_types::parameters::{
     PromiseBatchAction, PromiseCreateArgs, PromiseWithCallbackArgs,
 };
-use aurora_engine_types::types::AddressValidationError;
+use aurora_engine_types::types::{AddressValidationError, Fee};
 
 pub const ERR_NOT_ENOUGH_BALANCE_FOR_FEE: &str = "ERR_NOT_ENOUGH_BALANCE_FOR_FEE";
-pub const NO_DEPOSIT: Balance = 0;
+/// Indicate zero attached balance for promise call
+pub const ZERO_ATTACHED_BALANCE: Balance = 0;
+/// NEAR Gas for calling `fininsh_deposit` promise. Used in the `deposit` logic.
 const GAS_FOR_FINISH_DEPOSIT: NearGas = NearGas::new(50_000_000_000_000);
+/// NEAR Gas for calling `verify_log_entry` promise. Used in the `deposit` logic.
 // Note: Is 40Tgas always enough?
 const GAS_FOR_VERIFY_LOG_ENTRY: NearGas = NearGas::new(40_000_000_000_000);
 
+/// Admin control flow flag indicates that all control flow unpause (unblocked).
 pub const UNPAUSE_ALL: PausedMask = 0;
+/// Admin control flow flag indicates that the deposit is paused.
 pub const PAUSE_DEPOSIT: PausedMask = 1 << 0;
+/// Admin control flow flag indicates that withdrawal is paused.
 pub const PAUSE_WITHDRAW: PausedMask = 1 << 1;
 
+/// Eth-connector contract data. It's stored in the storage.
+/// Contains:
+/// * connector specific data
+/// * Fungible token data
+/// * paused_mask - admin control flow data
+/// * io - I/O trait handler
 pub struct EthConnectorContract<I: IO> {
     contract: EthConnector,
     ft: FungibleTokenOps<I>,
     paused_mask: PausedMask,
     io: I,
 }
 
-/// eth-connector specific data
+/// Connector specific data. It always should contain `prover account` -
 #[derive(BorshSerialize, BorshDeserialize)]
 pub struct EthConnector {
+    /// It used in the Deposit flow, to verify log entry form incoming proof.
     pub prover_account: AccountId,
+    /// It is Eth address, used in the Deposit and Withdraw logic.
     pub eth_custodian_address: EthAddress,
 }
 
-/// Token message data
-#[derive(BorshSerialize, BorshDeserialize)]
-#[cfg_attr(not(target_arch = "wasm32"), derive(Debug))]
-pub enum TokenMessageData {
-    Near(AccountId),
-    Eth {
-        receiver_id: AccountId,
-        message: String,
-    },
-}
-
-/// On-transfer message
+/// On-transfer message. Used for `ft_transfer_call` and  `ft_on_transfer` functions.
+/// Message parsed from input args with `parse_on_transfer_message`.
 pub struct OnTransferMessageData {
     pub relayer: AccountId,
     pub recipient: EthAddress,
-    pub fee: U256,
+    pub fee: Fee,
 }
 
 impl<I: IO + Copy> EthConnectorContract<I> {
-    pub fn get_instance(io: I) -> Self {
+    /// Init Eth-connector contract instance.
+    /// Load contract data from storage and init I/O handler.
+    /// Used as single point of contract access for various contract actions
+    pub fn init_instance(io: I) -> Self {
```

Comment:
Nit pick: I don't know if "initialize" is the right verb here. Maybe "create" would be better. I don't have strong feelings either way, but I don't there to be any confusion between this function and `init_contract` (which is only called once).

Reply:
I thought about it. I thought that "creation" is not an accurate name. 
We load previously created data. Initialization more accurately reflects the essence of what is happening. From this point of view, it would be more successful to rename `init_contract` to `create_contract`.